extensions/amp-carousel/0.1/slidescroll.js

```diff
@@ -598,8 +614,8 @@ export class AmpSlideScroll extends BaseSlides {
    * @private
    */
   hideRestOfTheSlides_(indexArr) {
-    const noOfSlides_ = this.noOfSlides_;
-    for (let i = 0; i < noOfSlides_; i++) {
+    const totalNumSlides = this.slides_.length;
```

Comment:
ah now i see what your problem is :) 

you should check 

min(this. noOfSlides_ , this.slides_.length);